integration-tests/src/main/resources/templates/HtmxApp/index.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Htmx Powered</title>
+  <link rel="icon" href="/favicon.ico" type="image/x-icon">
+  <script src="/htmx.js"></script>
+  <link rel="stylesheet" media="screen" href="/webjars/bootstrap/css/bootstrap.min.css">
+  <style>
+    input:read-only {
+      background-color: #e9ecef;
+    }
+  </style>
+</head>
+<body hx-headers='{"{inject:csrf.headerName}":"{inject:csrf.token}"}'>
+  <main class="container">
+
+    <h1>Htmx Powered</h1>
+    <div hx-get="{uri:HtmxApp.index}" hx-trigger="refreshList from:body">
+      {#fragment id="list"}
+      {#for contact in contacts}
+        <div class="row mb-4">
+          {#include HtmxApp/view /}
```

Comment:
OK, this is a very interesting use-case, and one I did not see coming. It appears here you're using a "toplevel" template as a mix of a tag and fragment. This is where the boundary between fragments, tags and includes is not well-defined.

At this location, I would much have preferred a `{#view contact/}` rather than an include. But you can't because it's not in `tags/view.html`.

Not sure what to think of this.

Reply:
in my opinion, using include for partials is totally fine, it is done in most other templating engines. 

What I do in general is use a `partials` directory to make it nice, I will do it because even if it's a test, some use them as source of inspiration.

Reply:
Also I don't think that tags are meant to be used as partials, they are meant to be used as snippets of reusable code (form, input, ...).

Reply:
So, what's the difference, in your opinion, between partials and tags? What I truly hate about includes is that they have the same pollution problem as C `#includes`, which is partially mitigated by the type-safe declaration, but when calling the include, you have zero visibility as to what parameters you're passing it. It's like it's sucking them from under you.

So, perhaps the difference here is that you only use a partial in one place, whereas you exect tags to be reused in many places?

Also, why is this partial not simply declared in the template it's used in, as a fragment? What's the rule for putting it inline in the main template vs moving it out to its own file?

Reply:
In this case, it is used both from the template and from the controller.
I changed a bit to make it clearer:
<img width="772" alt="image" src="https://github.com/quarkiverse/quarkus-renarde/assets/2223984/3c266476-2b11-4c68-a157-531c86585184">

In my mind at least, partials could be used in different places, but the primary goal for splitting the code and avoid huge pages. They are like fragments, but that you prefer having in a different file.

I haven't pushed yet, I have an issue with the gravatar (I am adding a test)

Reply:
> In this case, it is used both from the template and from the controller

Well, tags can also be declared in Java and used by controllers (in fact, we should always do this, otherwise we get no validation).

I'm wondering if we should not allow nested classes in `@CheckedTemplate`, so you could declare a `partials` class and it would automatically have the right path, relative to the parent class's path.

Reply:
IMO we shouldn't use tags as partials, but figure out how to have typesafety for partials (using includes)

Reply:
Well, the only difference between tags and includes is the syntax used to pass arguments to them. In the case of tags it's explicit, in the case of includes it's implicit.

If they have a `@CheckedTemplates` declaration, they both should be type-safe. Well, perhaps the call-sites are not type-checked, though, I think this is not implemented in either of them. https://github.com/quarkusio/quarkus/issues/21859

Reply:
So, I'm still not clear what you see as the difference between partials and tags? I've never understood why people think they're different.

Reply:
OK, but that's only a namespace difference.

If you viewed `view.html` as if it were a tag (don't change anything, just your perception, don't even move it from where it is), then instead of including it with:

```html
{#include HtmxApp/view /}
```

You could include it with:

```html
{#view contact/}
```

I'm pretty sure tags are just regular includes underneath.

Reply:
I don't think what you are suggesting would work, then it would be confusing next to using `it` (direct arguments). I would be more into something like `{#contact/view /}` in that case.

Reply:
Heh, the `{#tagname contact /}` already exists, it's a shortcut for `{#tagname contact=contact /}` and it works for more than just a single argument.